ENH: Display in progress popup when loading DICOM data

jcfr opened the pull request:
See https://discourse.slicer.org/t/slicer-custom-application-deployment-to-many-computers/18019/149?u=jcfr

```
Co-authored-by: Andras Lasso <lasso@queensu.ca>
Co-authored-by: Steve Pieper <pieper@isomics.com>
```

jcfr committed:
ENH: Display in progress popup when loading DICOM data

See https://discourse.slicer.org/t/slicer-custom-application-deployment-to-many-computers/18019/149?u=jcfr

Co-authored-by: Andras Lasso <lasso@queensu.ca>
Co-authored-by: Steve Pieper <pieper@isomics.com>

pieper commented on Modules/Scripted/QReads/QReads.py:
```diff
@@ -741,6 +750,11 @@ def loadDICOMDataDirectory(dicomDataDir):
 
           loadedNodeIDs.append(nodeID)
 
```
Would be good to add a try/except around this so that the dialog doesn't get wedged in an open state on error.  Instead a different message box could display that there was an error.

jcfr replied:
Agreed. A context manager would allow to nicely handle this.

jcfr committed:
ENH: Ensure "Import in progress dialog" is hidden in case of exception

This commit introduces the context manager "inProgressDialog" ensuring
the dialog is always hidden.

Co-authored-by: Steve Pieper <pieper@isomics.com>
